src/parser/tests/expressions_parser_tests.rs

```diff
@@ -1513,6 +1513,36 @@ fn assignment_to_null() {
     assert_eq!(ast_string, expected_ast);
 }
 
+#[test]
+fn assignment_to_number_with_implicit_and_explicit_plus_sign() {
+    let src = "
+        PROGRAM exp
+        x : DINT := 1;
+        y : DINT := +1;
+        END_PROGRAM
```

Comment:
this ST-code is not 100% correct. it should maybe look like this:
```st
PROGRAM exp
   VAR
      x : INT;
   END_VAR
   x := 1;
   x := +1;
END_PROGRAM
```

Reply:
Out of curiosity, would the following ST-code also be wrong because it's missing a `VAR` block?
```ST
PROGRAM exp 
x := NULL;
END_PROGRAM
``` 
(taken from the `assignment_to_null()` test)

Reply:
> Out of curiosity, would the following ST-code also be wrong because it's missing a `VAR` block?
> 
> ```smalltalk
> PROGRAM exp 
> x := NULL;
> END_PROGRAM
> ```
> 
> (taken from the `assignment_to_null()` test)

so technically this works, if x is declared somewhere as a global variable. When we do parse tests, we often dont have semantically correct cases: (e.g. we dont care if x would resolve to something meaningful or not).
your sourcode mixed a statement and a declaration, you cannot have declarations outside of VAR-blocks.

I was really surprised that the parser actually returned something 😄 . but if you look at your assignment, the parser created something very strange:
`Assignment{
left: "DINT",
right: "1" 
}`

so the parser thought that DINT is an identifer to a variable.

Reply:
Yeah that makes sense, thanks for clarifying.